__tests__/search.js

```diff
@@ -35,28 +35,53 @@ class MockVFSServiceProvider extends ServiceProvider {
     ];
   }
 }
+class MySearchAdapter {
+    constructor(core) {
+        this.core = core;
 
+    }
+
+    destroy() {
+    }
+
+    async init() {
+    }
+
+    async search(pattern) {
+        const list = [
+            {
+             path: 'jest:/test.jpg',
+             filename: pattern+'.jpg'   
+            }
+        ];
+        return list;
+    }
+}
 describe('Search', () => {
   let core;
   let search;
-
+  let options = {};
+  options = {
+       adapters: [MySearchAdapter],
+  };
   beforeAll(() => createInstance((c) => {
     c.register(MockVFSServiceProvider);
   }).then(c => core = c));
 
   afterAll(() => core.destroy());
 
   test('#constructor', () => {
-    search = new Search(core);
+    search = new Search(core,options);
   });
 
   test('#search', () => {
-    return search.search('hello')
+    return search.search('test')
       .then(result => {
-        expect(result).toMatchObject([{
-          path: 'jest:/hello.world',
-          filename: 'hello.world'
-        }]);
+	  console.log(result);
+        expect(result).toMatchObject([
```

Comment:
You can just put this into the other match. And make sure to keep indentation identical.

Reply:
> 



> You can just put this into the other match. And make sure to keep indentation identical.

The test is ok because it returns two items, one for the default search and the other for the custom search whose adapter is added.

Reply:
It returns *one* set of results. The `Search#search(pattern)` method combines all adapter results into one set, so this should be tested as such.

This is why it says `result` (singular).

Reply:
> It returns _one_ set of results. The `Search#search(pattern)` method combines all adapter results into one set, so this should be tested as such.
> 
> This is why it says `result` (singular).

I said the same